src/Input/Select/Select.tsx

```diff
@@ -15,346 +15,271 @@ import {
+  const handleFocus = React.useCallback(
+    (e: React.FocusEvent<HTMLInputElement>) => {
+      setIsFocus(true);
 
       if (onFocus !== undefined) {
         return onFocus(e);
       }
-    };
-
-    return listener;
-  };
+    },
+    []
+  );
 
-  handleChange = (
-    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
-  ) => {
-    const listener = (e: React.ChangeEvent<HTMLInputElement>) => {
-      const { children } = this.props;
+  // Should be called when the user types into the input
+  const handleChange = React.useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
       const childrenArray = React.Children.toArray(children) as Array<
         React.ReactElement<SelectItemProps>
       >;
 
-      this.setState({
-        selectedValue: e.target.value,
-        filterValue: childrenArray.filter(data =>
+      setSelectedValue(e.target.value);
+      setFilterValue(
+        childrenArray.filter(data =>
           data.props.children
             .toLowerCase()
             .includes(e.target.value.toLowerCase())
-        ),
-        cursor: 0,
-      });
-
-      if (onChange !== undefined) {
-        return onChange(e);
-      }
-    };
-
-    return listener;
-  };
-
-  handleClick = (
-    onOptionClick: (e: React.MouseEvent<HTMLLIElement, MouseEvent>) => void
-  ) => {
-    const listener = (e: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
-      const { children, onChange } = this.props;
-      this.setState({
-        selectedValue: e.currentTarget.innerText,
-        filterValue: React.Children.map(children, data => data),
-        isFocus: false,
-      });
+        )
+      );
+      setCursor(0);
 
       if (onChange !== undefined) {
-        const element = e.target as HTMLElement;
-        onChange(element.dataset.value);
+        return onChange(e.target.value);
       }
+    },
+    [children, onChange]
+  );
+
+  // Should be called when the user selects an option
+  const handleSelect = React.useCallback(() => {
+    const activeElement = getActiveElement();
+    const selectedValue = activeElement.innerText;
+    setSelectedValue(selectedValue || '');
+    setFilterValue(React.Children.map(children, data => data));
+    setIsFocus(false);
+
+    if (onChange !== undefined) {
+      onChange(selectedValue);
+    }
 
-      if (onOptionClick !== undefined) {
-        onOptionClick(e);
-      }
-    };
-
-    return listener;
-  };
-
-  handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
-    const { children, onChange, disableTyping } = this.props;
-    const { cursor, filterValue } = this.state;
-    const inputElement = e.target as HTMLInputElement;
-
-    if (disableTyping) {
-      e.preventDefault();
+    const activeElementIndex = Number(get(activeElement, 'dataset.id'));
+    const activeChild = filterValue[activeElementIndex];
+    const onOptionClick = get(activeChild, 'props.onOptionClick');
+    if (onOptionClick !== undefined) {
+      onOptionClick(selectedValue);
```

Comment:
I saw we were passing `event` as the parameter of `onOptionClick`. 

Did you check the usage of it in the `marketplace` and `employers` projects?

Reply:
Re-added it, shouldn't have been removed.